databricks/koalas/frame.py

```diff
@@ -8529,21 +8529,41 @@ def query(self, expr, inplace=False):
             * Internal columns that starting with __. are able to access,
               however, they are not supposed to be accessed.
             * This delegates to Spark SQL so the syntax follows Spark SQL
-            * If you want the exactly same syntax with pandas' you can work around
-              by using :meth:`DataFrame.map_in_pandas`. See the example below.
+            * If you want the exactly same syntax with pandas you can work around
+              by using :meth:`DataFrame.map_in_pandas`, when only the size of the
+              data is smaller than `compute.shortcut_limit`. See the example below.
 
+                >>> from databricks.koalas.config import set_option, reset_option, get_option
                 >>> df = ks.DataFrame([(1, 2), (3, 4), (5, 6)], columns=['A', 'B'])
                 >>> df
                    A  B
                 0  1  2
                 1  3  4
                 2  5  6
-                >>> num = 1
+                >>> # You can check the value of `compute.shortcut_limit` like the below.
+                ... get_option('compute.shortcut_limit')
+                1000
+                >>> # Since 1000 is larger than the size of the data(in this case, 3),
+                ... # we can work around with `map_in_pandas`.
+                ... num = 1
                 >>> df.map_in_pandas(lambda pdf: pdf.query('A > @num'))
```

Comment:
You should show an example with type hints. So it avoids short-cut, and shows a good practice example.

For the limitation by the referred variable not being serialized into Python worker sides, you can just add some descriptions such as :

> `map_in_pandas` is performed at workers in different nodes; however, currently the variable referred in pandas syntax is not serialized together. Therefore, in order to use such syntax,  you should make sure the variable is serialized by, for example, putting it within the closure.

```python
>>> def query_func(df):
...     num = 1
...     pdf.query('A > @num')
>>> df.map_in_pandas(lambda pdf: pdf.query('A > @num'))
```